packages/firestore/src/core/query.ts

```diff
@@ -631,6 +643,24 @@ export class KeyFieldFilter extends FieldFilter {
   }
 }
 
+/** Filter that matches on key fields within an array. */
+export class KeyFieldInFilter extends FieldFilter {
+  constructor(field: FieldPath, op: Operator, value: ArrayValue) {
+    super(field, op, value);
+  }
+
+  matches(doc: Document): boolean {
+    const arrayValue = this.value as ArrayValue;
```

Comment:
We can avoid this `as` cast by changing `value: ArrayValue` in the constructor definition to `public value: ArrayValue` (i.e. we override the public `value` property inherited from `FieldFilter` to be a more specific type (`ArrayValue` instead of `FieldValue`)).

But I see we are already doing this cast in `KeyFieldFilter`, `InFilter`, and `ArrayContainsFilter`.  Can you fix them too?